pkg/cmd/roachtest/README.md

```diff
@@ -6,38 +6,169 @@ separate) tool `roachprod`.
-1. Build a local binary of `roachtest`: `make bin/roachtest`
+- [Set up `roachprod`](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachprod/README.md), if you haven't already.
+- If you want to run tests that require an Enterprise license, add `COCKROACH_DEV_LICENSE=<key>` to your path. Cockroach Labs employees can find one in [Slack](https://cockroachlabs.slack.com/archives/CJX8V9SJ2/p1597348203368800?thread_ts=1597348076.367800&cid=CJX8V9SJ2).
+- For Cockroach Labs employees, if your local username does not match your
+  gcloud/`@cockroachlabs.com` username, add
+`ROACHPROD_USER=<your_crl_username>` to your environment, or add the
+`--user=$CRL_USERNAME` flag to the `roachtest run` invocations. Non-Cockroach
+Labs employees will want to set `--gce-project` (and maybe `--user` too) as the
+default project will not be accessible to them.
 
 # Usage
 
-You'll usually use `roachtest run` and specify a single test.
+`roachtest` has the list of tests it supports compiled in. They can be listed via:
 
-For Cockroach Labs employees, if your local username does not match
-your gcloud/`@cockroachlabs.com` username, you'll need to add
-`--user=$CRL_USERNAME` to the `roachtest` command. For non-Cockroach
-Labs employees, set `--gce-project` (and maybe `--user` too).
+```
+$ roachtest list
+acceptance/build-analyze [server]
+acceptance/build-info [server]
+acceptance/cli/node-status [server]
+[...]
+```
+
+## Getting the binaries
+
+To run a test, the `roachtest run` command is used. Since a test typically
+deploys a CockroachDB cluster, `roachtest` needs to know which cockroach binary
+to use. It also needs to know where to find `roachprod`, and also generically
+requires the `workload` binary which is used by many tests to deploy load
+against the cluster. To that effect, `roachtest run` takes the `--cockroach`,
+`--roachprod`, and `--workload` flags. The default values for these are set up
+so that they do not need to be specified in the common case. Besides, when
+the binaries have nonstandard names, it is often more convenient to move them
+to the location `roachtest` expects (it will tell you) rather than to specify
+the flags. However, when multiple binaries are around, it might be preferrable
+to be explicit, to avoid accidentally running tests against against the wrong
+version of CockroachDB.
+
+### Building your own
+
+The `cockroach` and `workload` binaries need to be built for the architecture
+of the target cluster. In most cases, this means `amd64-linux-gnu`, i.e. OSX
+users have to either download the binary they need from somewhere (see below)
+or cross-compile the binary locally, which unfortunately takes a long time (due
+to the overhead of virtualization for Docker on Mac).
+
+As a general rule of thumb, if running against "real" VMs, and the version to
+test is checked out, the following will build the right binaries and put them
+where `roachtest` will find them:
```

Comment:
Might be worth mentioning somewhere around here that on macOS, `roachtest` will prefer `bin/workload` if present (a macOS binary) and fall back to `bin.docker_amd64/workload` (the cross-compiled Linux binary) if it does not exist. This is kind of non-discoverable, and tripped me up a couple of times when I first started out with `roachtest` -- macOS users should either make sure `bin/workload` does not exist, or pass the path explicitly.

Reply:
Thanks, I added a note.

Reply:
Did you forget to push?

Reply:
No, still working on the PR. Sorry, should've staged that comment. Still getting used to using vanilla Github reviews more :-)
I'll ping when it's ready.

Reply:
Updated.